File: typescript/docs_code_snippets/knowledge-add.ts
import { Opper } from "opperai";

const opper = new Opper();

// setup
const kb = await opper.knowledge.create({ name: `docs-snippet-${Date.now()}` });
const kbId = kb.id;
// /setup

// --- docs ---
await opper.knowledge.add(kbId, {
  content: "TypeScript is a typed superset of JavaScript.",
  metadata: { source: "docs", topic: "typescript" },
});
// --- /docs ---

console.log("Document added");
// cleanup
await opper.knowledge.delete(kbId);

File: python/docs_code_snippets/knowledge-add.py
import time
from opperai import Opper

opper = Opper()

# Setup
kb = opper.knowledge.create(name=f"docs-snippet-{int(time.time())}")

# --- docs ---
opper.knowledge.add(
    kb.id,
    content="TypeScript is a typed superset of JavaScript.",
    metadata={"source": "docs", "topic": "typescript"},
)
# --- /docs ---

# Cleanup
opper.knowledge.delete(kb.id)
